src/users/service/userService.ts

```diff
@@ -17,86 +23,77 @@ export class UsersService {
   @Inject()
   usersRepository: UserRepository;
   async create(createUserDto: CreateUserDto): Promise<User> {
-
+    const auth = await dbAuth.createUser({
+      email: createUserDto.email,
+      password: createUserDto.password
+    });
+    createUserDto.authUid = auth.uid;
+    createUserDto.avatar = process.env.AVATAR_URL;
     const user = await this.usersRepository.create(createUserDto);
     if (!user) {
-      throw new NotFoundException(`how have wrong schema`);
+      throw new NotFoundException(`you have wrong schema`);
     }
     return user;
-
   }
 
   findAll(): Promise<Array<User>> {
     return this.usersRepository.findAll();
   }
 
-  async findOne(id: number): Promise<User> {
-    const user = await this.usersRepository.findOne(id);
+  async findOne(authUid: string): Promise<User> {
+    const user = await this.usersRepository.findOne(authUid);
     if (!user) {
-      throw new NotFoundException(`User with ID=${id} not found`);
+      throw new NotFoundException(`User with ID=${authUid} not found`);
     }
     return user;
   }
 
-   findOneEmail(condition: string): Promise<User> {
-    return this.userRepository.findOne({
-      where: [
-        { email: condition}
-      ],});
-  }
-
   update(id: number, updateUserDto: UpdateUserDto): Promise<User> {
-    const user=this.usersRepository.update(id, updateUserDto);
+    const user = this.usersRepository.update(id, updateUserDto);
     if (!user) {
       throw new NotFoundException(`User with ID=${id} not found`);
     }
     return user;
   }
 
-  remove(id: number): Promise<User> {
-    const user =this.usersRepository.remove(id);
-    if (!user) {
-      throw new NotFoundException(`User with ID=${id} not found`);
-    }
-    return user
-  }
-
-  async resetFindeOne(id: number,pass:string): Promise<User> {
-    const user = await this.userRepository.preload({
-      id: id,
-      password:pass
-    });
+  async remove(id: number, uid: string): Promise<User> {
+    const user = await this.usersRepository.remove(id, uid);
     if (!user) {
       throw new NotFoundException(`User with ID=${id} not found`);
     }
-    return this.userRepository.save(user);
-
+    return user;
```

Comment:
Can we ensure what's returning remove function? Maybe we can wrap it with try catch